packages/concerto-core/lib/decoratorutil.js

```diff
@@ -0,0 +1,409 @@
+     * @returns {Array<Object>} - the collection of decorator command sets
+     * @private
+     */
+    static transformNonVoabularyDecorators(dcsObjects, DCS_VERSION, namespace, decoratorData){
+        const {name, version} = ModelUtil.parseNamespace(namespace);
+        const nameOfDcs = name;
+        const versionOfDcs = version;
+        if (dcsObjects?.length > 0){
+            const dcmsForNamespace = {
+                '$class': `org.accordproject.decoratorcommands@${DCS_VERSION}.DecoratorCommandSet`,
+                'name': nameOfDcs,
+                'version': versionOfDcs,
+                'commands': dcsObjects
+            };
+            decoratorData.push(dcmsForNamespace);
+        }
+        return decoratorData;
+    }
+    /**
+     * Transforms the collected vocabularies into proper vocabulary command sets
+     * @param {Array<Object>} vocabObject - the collection of collected vocabularies
+     * @param {string} locale - the locale of vocabulary
+     * @param {string} namespace - the current namespace
+     * @param {Array<Object>} vocabData - the collection of existing vocabularies command sets
+     * @returns {Array<Object>} - the collection of vocabularies command sets
+     * @private
+     */
+    static transformVoabularyDecorators(vocabObject, locale, namespace, vocabData){
+        if (Object.keys(vocabObject).length > 0 ){
+            let strVoc = '';
+            strVoc = strVoc+`locale: ${locale}\n`;
+            strVoc = strVoc+`namespace: ${namespace}\n`;
+            strVoc = strVoc+'declarations:\n';
+            Object.keys(vocabObject).forEach(decl =>{
+                if (vocabObject[decl].term){
+                    strVoc += `  - ${decl}: ${vocabObject[decl].term}\n`;
+                    const otherProps = Object.keys(vocabObject[decl]).filter((str)=>str!=='term' && str!=='propertyVocabs');
+                    otherProps.forEach(key =>{
+                        strVoc += `    ${key}: ${vocabObject[decl][key]}\n`;
+                    });
+                }
+                if (vocabObject[decl].propertyVocabs && Object.keys(vocabObject[decl].propertyVocabs).length>0){
+                    if (!vocabObject[decl].term){
+                        strVoc += `  - ${decl}: ${decl}\n`;
+                    }
+                    strVoc += '    properties:\n';
+                    Object.keys(vocabObject[decl].propertyVocabs).forEach(prop =>{
+                        strVoc += `      - ${prop}: ${vocabObject[decl].propertyVocabs[prop].term}\n`;
+                        const otherProps = Object.keys(vocabObject[decl].propertyVocabs[prop]).filter((str)=>str!=='term');
+                        otherProps.forEach(key =>{
+                            strVoc += `        ${key}: ${vocabObject[decl].propertyVocabs[prop][key]}\n`;
+                        });
+                    });
+                }
+            });
+            vocabData.push(strVoc);
+        }
+        return vocabData;
```

Comment:
If you are returning `vocabData`, then you don't need to accept it as an argument. If you are mutating as a side-effect, you don't need to return it.